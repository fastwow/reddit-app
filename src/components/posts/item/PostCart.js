import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Image, Text, View, TouchableOpacity} from 'react-native';
import styles from './styles';

class PostCart extends Component {

  constructor(props) {
    super(props);
    this.info = this.props.info;
    this.viewItem = this.props.viewItem.bind(this, this.info);
  }

  render() {
    return (<TouchableOpacity style={styles.listItem} onPress={this.viewItem}>
        <View style={styles.imageContainer}>
          <Image
            style={styles.image}
            source={{
              uri: this.info.thumbnail ? this.info.thumbnail : 'https://ipsumimage.appspot.com/70x70',
            }}
          />
        </View>
        <View style={styles.titleContainer}>
          <Text style={styles.title}>
            {this.info.title}
          </Text>
        </View>
      </TouchableOpacity>
    );
  }
}

PostCart.propTypes = {
  info: PropTypes.object.isRequired,
  viewItem: PropTypes.func.isRequired,
};

export default PostCart;
