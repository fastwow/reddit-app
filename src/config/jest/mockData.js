export const postsResponse = {
  children: [
    {
      data: {
        url: 'https://v.redd.it/1zhxx58rhp421',
        id: '98vj9e',
        thumbnail: 'https://b.thumbs.redditmedia.com/NFJ5zzCUnxTgAl71EkjSSCePc7wtcCLqTdbYgrwJkdU.jpg',
        title: 'Blinks of affection',
      },
    },
    {
      data: {
        url: 'https://i.imgur.com/GsoJi0d.gifv',
        id: 'a6q94b',
        thumbnail: 'https://b.thumbs.redditmedia.com/cjYmtjKiy9OT1y43RFMuTkcH_WmTM66RsSuN7c6xkhA.jpg',
        title: 'This church looks like a gingerbread house.',
      },
    },
  ],
};

export const posts = postsResponse.children.map(c => c.data);
