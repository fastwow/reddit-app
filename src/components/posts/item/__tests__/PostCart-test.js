import React from 'react';
import renderer from 'react-test-renderer';
import PostCart from '../PostCart';
import {posts} from '../../../../config/jest/mockData';
import jest from 'jest-mock';

test('PostCart renders correctly', () => {
  const tree = renderer.create(
    <PostCart
      viewItem={jest.fn()}
      item={posts[0]}/>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('PostCart renders with image placeholder', () => {
  const post = posts[0];
  post.thumbnail = 'default';
  const tree = renderer.create(
    <PostCart
      viewItem={() => {}}
      item={post}/>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
