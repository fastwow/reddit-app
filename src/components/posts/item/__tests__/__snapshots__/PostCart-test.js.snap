// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCart renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#d6d7da",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 6,
    }
  }
>
  <View
    style={
      Object {
        "padding": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://b.thumbs.redditmedia.com/NFJ5zzCUnxTgAl71EkjSSCePc7wtcCLqTdbYgrwJkdU.jpg",
        }
      }
      style={
        Object {
          "height": 70,
          "width": 70,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "margin": 6,
          "textAlign": "left",
        }
      }
    >
      Blinks of affection
    </Text>
  </View>
</View>
`;

exports[`PostCart renders with image placeholder 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#d6d7da",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 6,
    }
  }
>
  <View
    style={
      Object {
        "padding": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://ipsumimage.appspot.com/70x70",
        }
      }
      style={
        Object {
          "height": 70,
          "width": 70,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "margin": 6,
          "textAlign": "left",
        }
      }
    >
      Blinks of affection
    </Text>
  </View>
</View>
`;
